crates/oxc_span/src/source_type.rs

```diff
@@ -182,14 +181,15 @@ impl SourceType {
                 )
             })?;
 
-        let is_definition_file = file_name.ends_with(".d.ts")
-            || file_name.ends_with(".d.mts")
-            || file_name.ends_with(".d.cts");
-
         let language = match extension {
             "js" | "mjs" | "cjs" | "jsx" => Language::JavaScript,
-            "ts" | "mts" | "cts" | "tsx" => Language::TypeScript { is_definition_file },
-            _ => unreachable!(),
+            "ts" if file_name.ends_with(".d.ts") => Language::TypeScriptDefinition,
```

Comment:
This is 3 string matches, can be reduced to strip start + match?

Reply:
My intent with moving the `ends_with` checks was to avoid testing `ends_with(".d.cts")` when `extension` is `ts`, as it can never match.

Reply:
Oh, never mind, I thought I had something more clever 😅

Reply:
Yes it's not completely optimal. But as Arnaud said on #2677, maybe this should have further changes anyway.